### PR TITLE
support integer id

### PR DESCRIPTION
**Pull request by nevermore3**

### CODE REVIEW

File: src/common/thrift/ThriftTypes.h [hunk 1]
```diff
@@ -27,7 +27,7 @@ using Port = int32_t;
 //
 // If the length of a given id is shorter than the specified length, '\0' will
 // be appended to the end
-using VertexID = std::string;
+using VertexID = Value;
```
Comment (Shylock-Hg) [inline src/common/thrift/ThriftTypes.h:30]: Add comment to specify the VertexID could only be INT or STRING of VALUE

File: src/common/interface/common.thrift [hunk 2]
```diff
@@ -147,14 +146,14 @@ struct Tag {
 
 
 struct Vertex {
-    1: VertexID vid,
+    1: Value     vid,
```
Comment (Shylock-Hg) [inline src/common/interface/common.thrift:149]: Why don't use `VertexID`?

File: src/common/interface/common.thrift [hunk 3]
```diff
@@ -147,14 +146,14 @@ struct Tag {
...
     2: list<Tag> tags,
 } (cpp.type = "nebula::Vertex")
 
 
 struct Edge {
-    1: VertexID src,
-    2: VertexID dst,
+    1: Value src,
```
Comment (Shylock-Hg) [inline src/common/interface/common.thrift:155]: ditto

File: src/common/thrift/ThriftTypes.h [hunk 4]
```diff
@@ -27,7 +26,7 @@ using Port = int32_t;
 //
 // If the length of a given id is shorter than the specified length, '\0' will
 // be appended to the end
-using VertexID = Value;
+using VertexID = std::string;
```
Comment (Shylock-Hg) [inline src/common/thrift/ThriftTypes.h]: Why change this?

File: src/common/clients/storage/GraphStorageClient.cpp [hunk 5]
```diff
@@ -504,154 +504,78 @@ GraphStorageClient::lookupAndTraverse(GraphSpaceID space,
...
-                DCHECK_EQ(Value::Type::STRING, e.key.src.type());
-                DCHECK_EQ(Value::Type::STRING, e.key.dst.type());
-                return e.key.src.getStr();
-            };
-    } else {
+    if (vidType != meta::cpp2::PropertyType::INT64 ||
+        vidType != meta::cpp2::PropertyType::FIXED_STRING) {
         return Status::Error("Only support integer/string type vid.");
     }
+    auto cb = [](const cpp2::NewEdge& e) -> const Value& { return e.key.src; };
```
Comment (laura-ding) [inline src/common/clients/storage/GraphStorageClient.cpp:548]: the graphd has checked it, I think here can delete

File: src/common/clients/storage/GraphStorageClient.cpp [hunk 6]
```diff
@@ -504,154 +504,78 @@ GraphStorageClient::lookupAndTraverse(GraphSpaceID space,
...
-                return mutableEK.src.getStr();
-            };
-    } else if (vidType == meta::cpp2::PropertyType::FIXED_STRING) {
-        cb = [](const cpp2::EdgeKey& eKey) -> const VertexID& {
-                DCHECK_EQ(Value::Type::STRING, eKey.src.type());
-                DCHECK_EQ(Value::Type::STRING, eKey.dst.type());
-                return eKey.src.getStr();
-            };
-    } else {
+    if (vidType != meta::cpp2::PropertyType::INT64 ||
```
Comment (laura-ding) [inline src/common/clients/storage/GraphStorageClient.cpp:559]: ditto

File: src/common/interface/common.thrift (hunk 2 above)
Comment (nevermore3) [reply src/common/interface/common.thrift:149]: storage treat vertexID as std::string,

File: src/common/function/FunctionManager.cpp [hunk 7]
```diff
@@ -1225,7 +1226,14 @@ FunctionManager::FunctionManager() {
             if (!args[0].isEdge()) {
                 return Value::kNullBadType;
             }
-            return args[0].getEdge().src;
+            auto& val = args[0].getEdge().src;
```
Comment (laura-ding) [inline src/common/function/FunctionManager.cpp:1229]: no need modify

File: src/common/function/FunctionManager.cpp [hunk 8]
```diff
@@ -1237,7 +1245,14 @@ FunctionManager::FunctionManager() {
             if (!args[0].isEdge()) {
                 return Value::kNullBadType;
             }
-            return args[0].getEdge().dst;
+            auto& val = args[0].getEdge().dst;
```
Comment (laura-ding) [inline src/common/function/FunctionManager.cpp:1248]: ditto

**Review by yixinglu — APPROVED** [review]

**Review by CPWstatic — APPROVED** [review]

**Pull request merged**